akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/common/ElasticDAO.scala

```diff
@@ -0,0 +1,174 @@
+                                         typeName: String,
+                                         where: Option[QueryDefinition] = None,
+                                         sort: Option[FieldSortDefinition] = None,
+                                         limit: Option[Int] = None)
+
+  case class DataRetrieved(
+                            index: String,
+                            typeName: String,
+                            data: RichSearchResponse,
+                            originalRQ: RetrieveAllFromIndexSorted
+                          )
+
+  case class ErrorRetrievingData(
+                                  error: Exception,
+                                  index: String,
+                                  typeName: String,
+                                  originalRQ: RetrieveAllFromIndexSorted
+                                )
+
+  object InitElasticDAO
+  case class ProblemConnectingWithES(e: Exception)
+  object ReadyToListen
+
+  private val settings = Settings.builder()
+    .put("cluster.name", TapiporlaConfig.ElasticSearch.clusterName)
+    .put("xpack.security.user", TapiporlaConfig.ElasticSearch.authUser)
+    .build()
+
+  val client = XPackElasticClient(settings, ElasticsearchClientUri(TapiporlaConfig.ElasticSearch.endpoint))
+
+}
+
+
+trait ElasticDAO extends Actor with ActorLogging with Stash{
+
+  def indexCreation: CreateIndexDefinition //Index to be created (if already exists we don't check if the mappings are similar)
+
+  self ! InitElasticDAO
+
+  override def receive = initial
+
+  def initial: Receive = {
+
+    case InitElasticDAO =>
+      client.execute(indexCreation) map { _ =>
+        ReadyToListen
+      } recover {
+        case CausedBy(ex: ResourceAlreadyExistsException) => //TODO: Change to see how to check if index already exists
+          ReadyToListen
+        case e: Exception =>
+          ProblemConnectingWithES(e)
+
+      } pipeTo (self)
+
+
+    case ProblemConnectingWithES(e) =>
+      log.error(s"Error initializing ElasticSearch DAO ${self.path.name} :$e. Trying again in 30 seconds")
+      context.system.scheduler.scheduleOnce(30 seconds, self, InitElasticDAO)
+
+
+    case ReadyToListen =>
+      unstashAll()
+      context.become(readyToProcess)
+
+    case _ =>
+      stash()
+  }
+
+
+  def readyToProcess: Receive = {
+
+    case SaveInIndex(index, typeName, jsonDocs) =>
+      client.execute {
+        bulk (
+          jsonDocs.map(x => indexInto(index / typeName).doc(x.json))
+        ) refresh(RefreshPolicy.WAIT_UNTIL)
+      } map { _=>
+        log.info(s"Completed to save in ES index $index type $typeName: ${jsonDocs.length} docs")
+        DataSavedConfirmation(index, typeName, jsonDocs)
+      } recover {
+        case e: Exception =>
+          log.error(s"Impossible to save in index $index $typeName ${jsonDocs.length} docs: $e")
+          ErrorSavingData(e, index, typeName, jsonDocs)
+      } pipeTo(sender)
```

Comment:
if you use infix notation, I would'nt wrap ```sender``` between parentheses.
```scala
}.pipeTo(sender)
```
or
```scala
} pipeTo sender
```

Reply:
:+1:  totally agree